torch/distributed/tensor/placement_types.py

```diff
@@ -173,7 +177,11 @@ def _shard_tensor(
 
         mesh_dim_local_rank = my_coordinate[mesh_dim]
         output = torch.empty_like(scatter_list[mesh_dim_local_rank])
```

Comment:
It looks when `src_data_rank is None`, `_shard_tensor` would return a tensor initiated using `empty_like`. This behavior is a bit different from what I'd expect of "using local data" -- e.g. if `distribute_tensor` is called on a tensor with replicate data on each rank, I would have expected the semantics of `Shard` to be kept even if `src_data_rank is None`. But maybe this is not an important use case.

Reply:
Good catch! Updated so that it respects the "local data" semantic in the doc :)